cli/src/main/java/de/jplag/cli/JPlagVersionChecker.java

```diff
@@ -0,0 +1,91 @@
+package de.jplag.cli;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Optional;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.jplag.JPlag;
+import de.jplag.reporting.reportobject.model.Version;
+
+/**
+ * Handles the check for newer versions.
+ */
+public class JPlagVersionChecker {
+    private static final String API_URL = "https://api.github.com/repos/jplag/JPlag/releases";
+    private static final Logger logger = LoggerFactory.getLogger(JPlagVersionChecker.class);
+    private static final String EXPECTED_VERSION_FORMAT = "v\\d\\.\\d\\.\\d+";
+    private static final String WARNING_UNABLE_TO_FETCH = "Unable to fetch version information. New version notification will not work.";
+    private static final String NEWER_VERSION_AVAILABLE = "There is a newer version ({}) available. You can fetch the newest version here: https://github.com/jplag/JPlag/releases.";
```

Comment:
Imo `download` would be a more fitting word than `fetch`, but i guess that is just preference